packages/client/src/components/project/input/Buttons.tsx

```diff
@@ -30,20 +35,56 @@ const START_TRAINING = gql`
   }
 `;
 
-export function StartButton(props: { id: string }): ReactElement {
+export function StartButton(props: { id: string; selected: GetProjectData_project_datasets[] }): ReactElement {
   const classes = useStyles();
   const [startTraining] = useMutation<StartTraining, StartTrainingVariables>(START_TRAINING);
   const [starting, setStarting] = useState(false);
 
+  const parameters = useQuery<GetHyperparameters, GetHyperparametersVariables>(GET_HYPERPARAMETERS, {
+    variables: {
+      id: props.id
+    }
+  });
+
+  const dockerState = useQuery<GetDockerState>(GET_DOCKER_STATE, { pollInterval: 5000 });
+  const datasets = useQuery<GetDatasets, GetDatasets>(GET_DATASETS);
+  const datasetNames = props.selected.map((dataset) => dataset.name);
+
   const handleClick = () => {
-    startTraining({ variables: { id: props.id } });
-    setStarting(true);
+    if (
+      !(
+        (parameters.data?.project?.epochs || 0) <= 0 ||
+        (parameters.data?.project?.batchSize || 0) <= 0 ||
+        (parameters.data?.project?.evalFrequency || 0) <= 0 ||
+        (parameters.data?.project?.percentEval || 0) <= 0
+      ) &&
+      datasetNames.length !== 0
+    ) {
+      startTraining({ variables: { id: props.id } });
+      setStarting(true);
+    }
   };
 
-  const { data, loading, error } = useQuery<GetDockerState>(GET_DOCKER_STATE, { pollInterval: 5000 });
-  if (loading) return <p>connecting to trainer</p>;
-  if (error) return <p>cant connect to trainer</p>;
-  if (data?.dockerState === DockerState.TRAIN_PULL) return <p>no train image yet</p>;
+  if (
+    (parameters.data?.project?.epochs || 0) <= 0 ||
+    (parameters.data?.project?.batchSize || 0) <= 0 ||
+    (parameters.data?.project?.evalFrequency || 0) <= 0 ||
+    (parameters.data?.project?.percentEval || 0) <= 0
+  ) {
+    return <Button> Invalid Parameters </Button>;
+  }
+
+  if (datasetNames.length === 0) {
+    return <Button> Requires Dataset </Button>;
```

Comment:
Be sure to set the styling of these buttons to be disabled so users don't try to click them.